### ISSUE TITLE
[Bug] Connectivity.NetworkAccess gives wrong network state (None) on Android 5.1 Lollipop

### ISSUE THREAD

**Issue by goedware** [xamarin/Essentials#911]
### Description
On Android 5.1 (Lollipop)` Connectivity.NetworkAccess` returns **None** while there are networks available. Essentials uses a` Platform.HasApiLevel(BuildVersionCodes.Lollipop) `check but it should be a `Platform.HasApiLevel(BuildVersionCodes.Marshmellow)` check before it uses the native Android function **getAllNetworks** of the ConnectivityService. **getAllNetworks** is API 23 (Marshmellow) and **getAllNetworkInfo** is still API 22 (lollipop). The version check is one version low. 
[https://developer.android.com/reference/android/net/ConnectivityManager#getAllNetworkInfo()](url) 

### Steps to Reproduce

1.  Add Xamarin Essentials to Android project
2.  Set Android minimum API to 22 Lollipop
3.  Retrieve value Connectivity.NetworkAccess
4.  Call getAllNetworkInfo in Android native code and you see network available

### Expected Behavior
On Android version 5.1 `Connectivity.NetworkAccess` should return the correct network status by using native `getAllNetworkInfo `method instead of `getAllNetworks`

### Actual Behavior
On Android version 5.1 `Connectivity.NetworkAccess` should returns the incorrect network status by using native `getAllNetworks `method instead of `getAllNetworkInfo`

### Basic Information

- Version with issue:  1.3.0
- Last known good version: ---
- IDE: Visual Studio 2019
- Platform Target Frameworks:  
  - Android: Android 5.1 SDK API 22

**Comment by jamesmontemagno** [comment]
So GetAllNetwork was introduced in API 21, which is why we use it: https://developer.android.com/reference/android/net/ConnectivityManager#getAllNetworks()

If you are able to run this specific code on your device: https://github.com/xamarin/Essentials/blob/master/Xamarin.Essentials/Connectivity/Connectivity.android.cs#L53-L89 and determine why it isn't returning a status on your device.

If you have a specific device that you are using that would be helpful.

**Comment by goedware** [comment]
It is a **Crestron** tablet device running Android 5.1.1. / `BuildVersionCodes.LollipopMr1` 
I also retrieved this version info using `DeviceInfo `and it matches the manufacturer data.

I attached 3 screenshots to verify my findings on this device. Using a test app attached to a download of the Essentials library.
1. Screenshot showing version information retrieved for this sample app
![Android Version](https://user-images.githubusercontent.com/6526600/66271593-2ebaa880-e860-11e9-8d08-6fa6c9e16136.PNG)
2. Screenshot showing the debug information using default Connectivity class of the Essentials library. That one shows an empty array of networks for `GetAllNetworks()`.
![GetAllNetworks](https://user-images.githubusercontent.com/6526600/66271594-2ebaa880-e860-11e9-93ea-4cc04f9c0212.PNG)
3. Screenshots showing the debug information using an edited Connectivity class (inserted a ! in the if statement) to use the obsolete method `manager.GetAllNetworkInfo()`. That one shows an array containing 3 network info objects.
![GetAllNetworkInfo](https://user-images.githubusercontent.com/6526600/66271592-2ebaa880-e860-11e9-857e-7d7bfc02ed44.PNG)

If you need more info please tell me and I will try to retrieve that information from my test application.

**Comment by jamesmontemagno** [comment]
Amazing! Yeah, i think this is device specific as I tested it on my lollipop device without issues, but I think we can easily tweak some stuff here to have a fallback if none are returned. I am testing it in my connectivit plugin and will have something for you soon.

**Comment by goedware** [comment]
I tested it on a mobile 5.1 device and 5.1 emulator and it was working correct as you would expect.
So I also think it is device specific and I happened to be thinking about asking if a fallback mechanism could be added for these specific devices too. If `getAllNetworks()` returns an empty array the library does an extra `getAllNetworkInfo()` just to be sure.
So if such a thing could be added, that would be very cool indeed 😃 👍!
